### PR TITLE
fix: handling invalid timestamp for adjust source

### PR DESCRIPTION
**Pull request by shrouti1507**
## What are the changes introduced in this PR?

Write a brief explainer on your code changes.
Screenshot 2024-11-11 at 3 40 40 PM

## What is the related Linear task?

Resolves INT-2873

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [x] My code follows the style guidelines of this project

- [x] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [x] Is the PR limited to 10 file changes?

- [x] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added in **new readability format**?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3866/test-report.html

**Review by yashasvibajpai — APPROVED** [review]

File: src/v0/sources/adjust/transform.js [hunk 1]
```diff
@@ -43,11 +43,18 @@ const processEvent = (inputEvent) => {
...
+    try {
+      const createdAt = Number(rawTimestamp);
+      const ts = new Date(createdAt * 1000).toISOString();
+      message.setProperty('originalTimestamp', ts);
+      message.setProperty('timestamp', ts);
+    } catch (error) {
+      throw new TransformationError(
+        `[Adjust] Invalid timestamp "${rawTimestamp}": ${error.message}`,
+      );
+    }
```
Comment (sanpj2292) [inline src/v0/sources/adjust/transform.js:56]: Write in it's own method

Comment (shrouti1507) [reply src/v0/sources/adjust/transform.js:56]: means?

Comment (shrouti1507) [reply src/v0/sources/adjust/transform.js]: addressed

File: src/v0/sources/adjust/utils.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+const { TransformationError } = require('@rudderstack/integrations-lib');
+
+const processTimestamp = (rawTimestamp) => {
```
Comment (sanpj2292) [inline src/v0/sources/adjust/utils.js:3]: Better name would be `getTimestamp` I believe. Let's make this minor change

Comment (yashasvibajpai) [reply src/v0/sources/adjust/utils.js:3]: should we keep in the outside utils? can this be useful in other destination built down the line?

Comment (sanpj2292) [reply src/v0/sources/adjust/utils.js:3]: Let's keep it in `adjust` itself. In future when there is a requirement, let's move it to the utils outside

**Review by sanpj2292 — APPROVED** [review]

**Review by yashasvibajpai — APPROVED** [review]

**Review by sanpj2292 — APPROVED** [review]

**Review by yashasvibajpai — APPROVED** [review]

**Pull request merged**